source/BannerlordCoopInstaller/Installer.cs

```diff
@@ -0,0 +1,264 @@
+﻿using Microsoft.Win32;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Reflection;
+using System.Resources;
+using System.Text;
+using System.Windows.Forms;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.ComTypes;
+
+namespace BannerlordCoopInstaller
+{
+    public partial class BannerlordCoopInstallerForm : Form
+    {
+
+        private Dictionary<string, ResourceFile> resourceDict;
+
+        /// <summary>
+        /// Check the BannerlordPath and ensure and executable exists. If not, disable the install button
+        /// </summary>
+        private void checkBannerLordPath()
+        {
+            if(File.Exists(this.BannerlordPath.Text + "\\bin\\Win64_Shipping_Client\\Bannerlord.exe"))
+            {
+                this.Install.Enabled = true;
+                this.Status.Text = "Ready to install";
+            }
+            else
+            {
+                this.Status.Text = "Select a Mount && Blade II: Bannerlord install path to continue";
+                this.Install.Enabled = false;
+            }
+        }
+
+        /// <summary>
+        /// Check if MB is installed (through the registry) - Only works in x64.
+        /// </summary>
+        /// <returns>Full path where MB is installed</returns>
+        private string GetBannerLordInstalledPath()
+        {
+
+            string registry_key = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall";
+            using (Microsoft.Win32.RegistryKey key = Registry.LocalMachine.OpenSubKey(registry_key))
+            {
+                foreach (string subkey_name in key.GetSubKeyNames())
+                {
+                    using (RegistryKey subkey = key.OpenSubKey(subkey_name))
+                    {
+                        string name = (string)subkey.GetValue("DisplayName");
+                        if (name != null && name.Equals("Mount & Blade II: Bannerlord"))
+                        {
+                            return subkey.GetValue("installlocation").ToString();
+                        }
+                    }
+                }
+            }
+            return null;
```

Comment:
Since we're using this to set the initial text, should probably use an empty string instead of null. Same difference but nicer imo
```suggestion
            return string.Empty;
```